app/src/main/java/org/olpc/xorobothash/MainActivity.java

```diff
@@ -0,0 +1,164 @@
+package org.olpc.xorobothash;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.os.AsyncTask;
+import android.support.v7.app.ActionBarActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ImageView;
+import android.util.Log;
+import android.widget.TextView;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.net.HttpURLConnection;
+
+
+
+public class MainActivity extends ActionBarActivity {
+
+    ImageView robotImageView;
+    String robotImagePath;
+
+    //For debugging
+    String tag = "XORoboHash";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        robotImageView = (ImageView) findViewById(R.id.robot_hash_image);
+        robotImagePath = this.getFilesDir() + File.pathSeparator + getString(R.string.robot_image_internal_filename);
+        getRobot();
+    }
+
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+
+    private void getRobot() {
+        Log.i (tag, "In getRobot()");
```

Comment:
Convention for spacing around parenthesis is to include a space after keywords (`for`, `while` etc), but not after function names in function calls or declarations. So remove this space.